### PR TITLE
Swift 3.0 Beta 1 Migration

### PR DESCRIPTION
**Pull request by mliberatore**
Progress on #123

📝 This is pointing at the new `swift-3.0` branch, not `develop`.

- Runs the migrator in Xcode 8 to update to Swift 3.0 syntax. This was all performed in c880af1.
- All other commits can / should be reviewed independently to fully understand the need for the changes. They fix leftover compiler errors, migrator-introduced force unwrapping that wasn't necessary, or sub-optimal optional chaining.

To keep this from being difficult to review, I’m deferring following of the [API Design Guidelines](https://swift.org/documentation/api-design-guidelines/) for introduced APIs to a separate branch and pull request.

### CODE REVIEW

File: PinpointKit/PinpointKit/Sources/BarButtonItem.swift [hunk 1]
```diff
@@ -18,8 +18,8 @@ extension UIBarButtonItem {
      - parameter action: The bar button item’s action.
      */
     convenience init(doneButtonWithTarget target: AnyObject?, font: UIFont, action: Selector) {
-        self.init(barButtonSystemItem: .Done, target: target, action: action)
+        self.init(barButtonSystemItem: .done, target: target, action: action)
         
-        setTitleTextAttributes([NSFontAttributeName: font], forState: .Normal)
+        setTitleTextAttributes([NSFontAttributeName: font], for: UIControlState())
```
Comment (mattbischoff) [inline PinpointKit/PinpointKit/Sources/BarButtonItem.swift]: This one still weirds me out, personally.

Comment (mliberatore) [inline PinpointKit/PinpointKit/Sources/BarButtonItem.swift]: There’s also [the option of `[]`](http://stackoverflow.com/questions/37800342/uicontrolstate-normal-is-unavailable)?

Comment (mattbischoff) [inline PinpointKit/PinpointKit/Sources/BarButtonItem.swift]: Wonder what the Swift team recommends?

Comment (mattbischoff) [inline PinpointKit/PinpointKit/Sources/BarButtonItem.swift]: Turns out, this is probably the best option.

**Pull request merged**